.github/workflows/build.yaml

```diff
@@ -22,29 +24,72 @@ jobs:
       matrix:
         include:
         - { arch: aarch64, runs-on: macos-14 }
-        - { arch: x86_64, runs-on: macos-13 }
+        - { arch: x86_64, runs-on: macos-12 }
     runs-on: ${{ matrix.runs-on }}
-    timeout-minutes: 40
+    timeout-minutes: 120
     steps:
-    - run: mkdir -p build
-    - run: brew install filemonitor ninja
-    - run: brew install --ignore-dependencies lima # Don't pull in qemu
     - uses: actions/checkout@v4
       with:
         persist-credentials: false
-    - run: brew install --build-from-source ./qemu.rb
+        fetch-depth: 0
+    - id: version
+      name: Calculate version
+      shell: bash
+      run: |
+        version=$(git describe --match 'v[0-9]*' --dirty='.m' --always --tags)
+        echo "VERSION=${version#v}" >>"$GITHUB_OUTPUT"
+    - name: Uninstall all brew packages and clear cache
+      run: |
+        set -o xtrace
+        brew list | xargs brew uninstall --force
+        brew autoremove
+        brew cleanup
+        rm -rf "$(brew --cache)"
+    - name: Build qemu and all prerequisites from source
+      run: |
+        set -o xtrace
+        brew tap-new rd/tap
+        cp qemu.rb "$(brew --repo rd/tap)/Formula/qemu.rb"
+        brew install --overwrite --build-from-source $(brew deps --include-build python@3.11)  python@3.11
+        brew install --overwrite --build-from-source $(brew deps --include-build python@3.12)  python@3.12
+        brew install --overwrite --build-from-source $(brew deps --include-build python@3.13)  python@3.13
```

Comment:
Why do we need three pythons? (A comment here would be useful.) Separate from the deps for qemu, at that.

Reply:
They are all needed to build build QEMU or it's prerequisites. They need to be installed manually because homebrew will just die on macOS 12 in this situation now.

I'll add a comment.